### ISSUE TITLE
Bug: Userfield of type "Select list multiple" + selecting no item will break editing userfield forms

### ISSUE THREAD

**Issue by FREE272** [grocy/grocy#2339]
TL;DR:
If you have a userfield of the type "Select list (multiple items can be selected)" with no items selected (i.e. containing data "None"), it will make it so when you click edit on a product, the fields for the following userfields will not contain any data. When you then click save it will override the data to "None" for the following userfields.

I initially encountered this error on Grocy version 3.3.2, then recently updated to Grocy version 4.0.3 where the issue still persists. The bug is also precent on pre-release-c8891236.

### Steps to reproduce:
**An example is set up here:**
https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/?nodemodata

_All steps will take place under the tab "Manage master data"_

**In the "Userfields" tab:** 
Add at least 3 userfields for entity "products", give them ascending sort numbers (i.e. 0, 1, 2), check the box "Show as column in tables".

One of the userfields has to be of the type "Select list (multiple items can be selected)", give that one a sort number in between the two other userfields (i.e. 1 if the other userfields have 0 and 2).
(The type of the other userfields does not matter.)

![Userfields](https://github.com/grocy/grocy/assets/142397937/a2e1463a-68a9-4dc8-8535-b94aea1ed8ff)

**In the "Products" tab:**
Add a product and fill in the mandatory fields with whatever.
On the bottom of the page, fill in all the userfields with appropriate data, except the userfield with the type "Select list (multiple items can be selected)" .
Click the blue button "Save & return to products" .

![Products](https://github.com/grocy/grocy/assets/142397937/36ae514b-2126-41c2-b387-93dee5a39ecf)

Now you should see that your product has the data you set in the userfields:
![Product saved](https://github.com/grocy/grocy/assets/142397937/865bea7b-a8ff-49d5-9cde-2b109b08cac3)

If you then click edit on the product you will see that the userfield(s) that come after the userfield with the type "Select list (multiple items can be selected)" are **empty**. If you now save the product, the userfields which previously contained data will be overwritten with empty values ("Null"). This is obviously not desirable as you have to fill in these userfields each time you edit a product to prevent them from being overwritten with "Null".

![Userfield empty](https://github.com/grocy/grocy/assets/142397937/7507b9df-5812-49a1-b80e-83825a1f5d70)

### The source of the problem
On the product edit page, I right-clicked and chose "Inspect Element" in my Firefox browser.
It reviled the following error message. When I clicked the error it showed me that there was a "TypeError" on line 146 in the file "userfieldsform.js?v=pre-release-c8891236"

Screenshot:
![Firefox error](https://github.com/grocy/grocy/assets/142397937/f5fa3b70-4477-43bc-9902-6a59d67efc2a)
Copied error text:
```
Uncaught TypeError: value is null
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:146
    each jQuery
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:136
    onreadystatechange https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/js/grocy.js?v=pre-release-c8891236:17
    Get https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/js/grocy.js?v=pre-release-c8891236:7
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:133
    <anonymous> https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/productform.js?v=pre-release-c8891236:260
userfieldsform.js:146:7
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:146
    each jQuery
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:136
    onreadystatechange https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/js/grocy.js?v=pre-release-c8891236:17
    (Async: EventHandlerNonNull)
    Get https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/js/grocy.js?v=pre-release-c8891236:7
    Load https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/components/userfieldsform.js?v=pre-release-c8891236:133
    <anonymous> https://test-f7u3jhc39yqbbtbiowlp5v.demo-prerelease.grocy.info/viewjs/productform.js?v=pre-release-c8891236:260
```

When I click the red squiggly line on the word input, at line 146, I get this message:

![TypeError](https://github.com/grocy/grocy/assets/142397937/21e947ef-5b0f-422d-8780-5577155da3b0)
